proto/tendermint/abci/types.proto

```diff
@@ -123,23 +123,31 @@ message RequestApplySnapshotChunk {
 }
 
 message RequestPrepareProposal {
-  bytes                   hash   = 1;
-  tendermint.types.Header header = 2 [(gogoproto.nullable) = false];
+  // the modified transactions cannot exceed this size.
+  int64 max_tx_bytes = 1;
   // txs is an array of transactions that will be included in a block,
   // sent to the app for possible modifications.
-  repeated bytes     txs                  = 3;
-  ExtendedCommitInfo local_last_commit    = 4 [(gogoproto.nullable) = false];
-  repeated Evidence  byzantine_validators = 5 [(gogoproto.nullable) = false];
-  // the modified transactions cannot exceed this size.
-  int64 max_tx_bytes = 6;
+  repeated bytes            txs                  = 2;
+  ExtendedCommitInfo        local_last_commit    = 3 [(gogoproto.nullable) = false];
+  repeated Evidence         byzantine_validators = 4 [(gogoproto.nullable) = false];
+  bytes                     hash                 = 5;
```

Comment:
This is the hash of the header right? Maybe add a comment here so applications don't think it's the app hash

Reply:
Wait it can't be the hash of the header because we don't know what the txs are. Then what is this?

Reply:
This hash currently is the hash of the candidate block if all of the Txs passed through were to be included. I'd imagine that frankly that's not that helpful at all. It may be better to just remove this since it's likely to change as a result of the `PrepareProposal` call

Reply:
Yeah I think it doesn't provide that much value and you'll get the agreed-upon header at `FinalizeBlock` so I would be for just removing it.

Reply:
However I would still add comments to the other references

Reply:
I agree with removing it. The App could use this field for sanity checks (making sure Tendermint isn't acting up), as e.g. _height_ can be used in other calls like _RequestExtendVote_.
But, in fairness, this seems overkill here. So I agree to removing it.

Reply:
Removed.